### PR TITLE
Feature/add metrics

### PR DESCRIPTION
**Pull request by werkaaa**
## What?
Prometheus configuration for dev and prod Grafana.
## How?
Added:
- `./prometheus/prod-prometheus.yml` and `./prometheus/dev-prometheus.yml` - Prometheus configuration (incl. scrape interval),
- `prod-docker-compose.yml` - configuration for running the app and local Prometheus,
- `dev-docker-compose.yml` - configuration for running the app, local Prometheus and local Grafana.
---------------------------------------------------------------------------------
## What?
Local Grafana configuration.
## Why?
This configuration will make it possible to see the metrics locally during development (without the need of using Grafana Cloud, where the metrics from VM will be sent).
## How?
Added:
- `grafana/dashboards/image_manager_dashboard.json` - local Grafana dashboard configuration,
- `grafana/provisioning/dashboard-config.yml` - Grafana configuration for automatic dashboard import.

### CODE REVIEW

File: dev-docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+version: "3"
```
Comment (Qizot) [inline dev-docker-compose.yml:1]: can we use more specific version? `3.8` seems to be the newest one

**Review by Qizot — CHANGES_REQUESTED** [review]
One small change in compose file needed. I guess production compose file will look quite different but I will change that accordingly when preparing deployment.

**Comment by remilvus** [comment]
# What
Various metrics exposed on /metrics endpoint  

# How
Modified:  
 - `application.yml` - added endpoint configuration and basic metric configuration
 - `ImageController` - added response timing and outbound traffic tracking
 - `ImageServiceImpl` - added counting cache misses and inbound (origin server) traffic tracking

File: dev-docker-compose.yml (hunk 1 above)
Comment (werkaaa) [reply dev-docker-compose.yml]: Done

**Review by Qizot — APPROVED** [review]

**Review by Przemyslaw5 — APPROVED** [review]

**Pull request merged**